client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -166,12 +178,23 @@ tie.factory('PythonCodeRunnerService', [
             code, '', [], [], [], errorTraceback,
             responseData[VARNAME_MOST_RECENT_INPUT]);
       } else if (responseData.results) {
-        return CodeEvalResultObjectFactory.create(
+        var codeEvalResult = CodeEvalResultObjectFactory.create(
             code, responseData.stdout,
             responseData.results[VARNAME_OBSERVED_OUTPUTS],
             responseData.results[VARNAME_BUGGY_OUTPUT_TEST_RESULTS],
             responseData.results[VARNAME_PERFORMANCE_TEST_RESULTS],
             null, null);
+        var question = CurrentQuestionService.getCurrentQuestion();
+        if (!question) {
```

Comment:
In what case would there be no question?

Reply:
Fixed specs to always have questions.